MuteToggler@jebeaudet.com/files/MuteToggler@jebeaudet.com/applet.js

```diff
@@ -192,12 +213,51 @@ MyApplet.prototype = {
         }
     },
 
+    evaluate_soundcard: function() {
+      // only use specific soundcard if searchstring is not empty
+      if (this.soundcard.trim() !== '')
+      {
+        // per default use first soundcard 
+        this.soundcard_id = "0";
+  
+        let soundcard_list_cmd = "cat /proc/asound/cards"; 
+        let [res, stdout] = GLib.spawn_command_line_sync(soundcard_list_cmd); 
+
+        if (res) {
+          // Split the result into lines 
+          let lines = stdout.toString().split('\n');
```

Comment:
Avoid `stdout.toString()`.

Use `to_string(stdout)` instead with:
```
const to_string = function(data) {
  const extraByteMap = [ 1, 1, 1, 1, 2, 2, 3, 0 ];
  var count = data.length;
  var str = "";

  for (var index = 0;index < count;)
  {
    var ch = data[index++];
    if (ch & 0x80)
    {
      var extra = extraByteMap[(ch >> 3) & 0x07];
      if (!(ch & 0x40) || !extra || ((index + extra) > count))
        return null;

      ch = ch & (0x3F >> extra);
      for (;extra > 0;extra -= 1)
      {
        var chx = data[index++];
        if ((chx & 0xC0) != 0x80)
          return null;

        ch = (ch << 6) | (chx & 0x3F);
      }
    }

    str += String.fromCharCode(ch);
  }

  return str;
}
```

Reply:
You forgot to declare to_string, didn't you?

Reply:
Well using Util.spawn_async gives stdout as string already.
And in line 13 there is already a to_string function imported, that is used in another place of the applet.